CONTRIBUTING.md

```diff
@@ -1,64 +1,96 @@
-## Managing vendor/
+## Contributing to ksonnet
 
-This project uses `govendor`.  To make things easier for reviewers,
-put updates to `vendor/` in a separate commit within the same PR.
+Thank you for taking the time to contribute to ksonnet! Before you submit any PRs, we encourage you to read the instructions for the [developer certificate of origin (DCO)](DCO-SIGNOFF.md), as well as the guidelines below:
 
-### Clean up removed/unnecessary dependencies
+* [Build](#build)
+* [Manage dependencies](#manage-dependencies)
+   * [Add a new dependency](#add-a-new-dependency)
+   * [Pin a dependency to a specific version](#pin-a-dependency-to-a-specific-version)
+   * [Clean up unnecessary dependencies](#clean-up-unnecessary-dependencies)
+* [Test](#test)
+* [Make a release](#make-a-release)
 
-This is a good one to run before sending a non-trivial change for
-review.
+### Build
 
+To build ksonnet locally (e.g. for open-source development), run the following in the root of your ksonnet repo:
 ```
-# See unused
-govendor list +unused
+make all
+```
+This indirectly makes the following targets:
+  * `make ks` - Compiles all the code into a `ks` binary
+  * `make docs` - Regenerates all the documentation for the ksonnet CLI, based on the code inside `cmd/`.
 
-# Remove unused
-govendor remove +unused
+*Troubleshooting*
+
+`make docs` relies on the `realpath` Linux utility. If you encounter a `realpath: command not found` error, and you are using OSX, run the following command:
+```
+brew install coreutils
 ```
+Running `make` again afterwards should work.
+
+### Manage dependencies
 
-### Add a new dependency
+This project uses `govendor` to manage Go dependencies. To make things easier for reviewers, put updates to `vendor/` in a separate commit within the same PR.
```

Comment:
I'd specifically call out the requirements of adding a vendored dependency. Perhaps something like (this is a really quick cut, feel free to discard):

```
This project uses `govendor` to manage dependencies. When you introduce a dependency to the ksonnet codebase, you need to:

1. Discuss the use case and need for the dependency with the maintainers by opening an issue.
1. After building agreement, vendor the dependency using `govendor`.
  1. Where possible, **please pin the repository to a specific stable version.** Please do not vendor HEAD if you can avoid it!
  1. **Please introduce a vendored dependency in a dedicated commit to make review easier!**
1. Write the code that exercises the dependency.
```